### DISCUSSION TITLE
請益關於Mac安裝步驟的一些疑問

### DISCUSSION THREAD

**Discussion by MiloCT** [GpointChen/FFXIVChnTextPatch-GP#380] [General]
MacOS在安裝的過程中有些疑問求解

1.專案本身 build.gradle 中不包含， com.univocity.parsers.csv 這個 package 需要 [下載]，並放到 專案目錄/libs 中。
請問下載下來的檔案是一個univocity-parsers-2.9.1.jar的檔案對嗎？
另外就是把檔案放入專案目錄/libs 請問這邊的專案目錄指的是gradle 6.x.x 的版本這個資料夾嗎？
如果是的話我在裡面只有找到gradle-6.8.3/lib 找不到libs這個資料夾

2.如不想下載此函式庫，打開 build.gradle 編輯下列區塊
請問要去哪裡打開build.gradle，我有上網google但也找不到 恕我愚笨 求解

3.Gradle編譯這個專案需要 gradle 6.x.x 的版本，可以[透過此連結下載](https://gradle.org/next-steps/?version=6.8.3&format=bin)。 請記得下載的路徑，並將解壓縮後資料夾裡的 bin 路徑加入環境變數 PATH 中。
請問把bin路徑加入環境變數的方式就是以下這個嗎？（我是Mac Intel)
設定 openjdk 的路徑 Mac (Intel)
export PATH="/usr/local/opt/openjdk@11/bin:$PATH"

4.最後一步使用gradle run會出現以下錯誤，我猜應該是我沒有把com.univocity.parsers.csv這個package步驟完成的關係？
* What went wrong:
Task 'run' not found in root project 'Milo'.

* Try:
Run gradle tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 401ms

以上幾個問題求解，第一次用這些軟體所以即使跟著步驟一步一步走還是發現似乎有些需要具備一些基本知識才行得通...

麻煩大神求解～感謝各位

**Comment by hydai** [comment]
Hi @MiloCT 
> 1.專案本身 build.gradle 中不包含， com.univocity.parsers.csv 這個 package 需要 [下載]，並放到 專案目錄/libs 中。
請問下載下來的檔案是一個univocity-parsers-2.9.1.jar的檔案對嗎？

是的

> 另外就是把檔案放入專案目錄/libs 請問這邊的專案目錄指的是gradle 6.x.x 的版本這個資料夾嗎？

專案目錄指的是「[FFXIVChnTextPatch-GP](https://github.com/GpointChen/FFXIVChnTextPatch-GP)」

> 2.如不想下載此函式庫，打開 build.gradle 編輯下列區塊
請問要去哪裡打開build.gradle，我有上網google但也找不到 恕我愚笨 求解

`build.gradle` 是一個在專案目錄內的檔案

> 3.Gradle編譯這個專案需要 gradle 6.x.x 的版本，可以[透過此連結下載](https://gradle.org/next-steps/?version=6.8.3&format=bin)。 請記得下載的路徑，並將解壓縮後資料夾裡的 bin 路徑加入環境變數 PATH 中。
請問把bin路徑加入環境變數的方式就是以下這個嗎？（我是Mac Intel)
設定 openjdk 的路徑 Mac (Intel)
export PATH="/usr/local/opt/openjdk@11/bin:$PATH"

`openjdk` 是給這個專案所使用的，設定路徑的方式同上

`gradle` 請參考這個段落進行設定 https://github.com/GpointChen/FFXIVChnTextPatch-GP/blob/master/docs/MACOS_BUILD.md#%E8%A8%AD%E5%AE%9A-gradle-%E7%9A%84%E8%B7%AF%E5%BE%91

> 4.最後一步使用gradle run會出現以下錯誤，我猜應該是我沒有把com.univocity.parsers.csv這個package步驟完成的關係？

應該不是，看起來你的 `gradle` 有設定到，你需要切換到專案目錄下面再執行 `gradle run`

**Comment by hydai** [comment]
Hi @MiloCT 
我在 https://github.com/GpointChen/FFXIVChnTextPatch-GP/discussions/342 直接提供了包裝好的 dmg 檔案，你可以下載後直接使用
